src/connectors/infinite-hits/connectInfiniteHits.ts

```diff
@@ -240,11 +241,14 @@ const connectInfiniteHits: InfiniteHitsConnector = (
         if (!hasShowPrevious) {
           return searchParameters;
         }
+
         const uiPage = uiState.page;
+
         if (uiPage) {
           return searchParameters.setQueryParameter('page', uiPage - 1);
         }
-        return searchParameters.setQueryParameter('page', 0);
+
+        return searchParameters.setQueryParameter('page');
```

Comment:
I wonder if it's worth explicitly passing `undefined` as second argument. Not sure how consistent we are about that in the code base

Reply:
Passing `undefined` would indeed be more explicit. We could perhaps enforce that by removing the `?` optional type in the added definition.

Reply:
IMO the argument `undefined` is useless and might be even counter-intuitive. Here, the intent is to omit the value on the `SearchParameters` which is what the call translate. With the value `undefined` we might expect that the final object will actually contain `undefined` which is not the case. I'm fine with both solution though, if you both agree that it's better I'll update.

Reply:
I'm not used to seeing setters without values, hence my reply. We could have a method in the helper called `resetQueryParameter(parameter: string)` which would make it clearer. In the end, it doesn't really matter.

Reply:
Ok, I will update the call. We should avoid creating methods on the helper that don't have "real" value. Otherwise, we have an API surface that became very broad. Like what we have at the moment. By multiplying the methods we might also introduce multiple ways to do the exact same operation, which doesn't help for consistency.

Reply:
https://github.com/algolia/instantsearch.js/pull/3797/commits/cb6ea7153e18d948620a65a902dcda91642b40e8, https://github.com/algolia/instantsearch.js/pull/3797/commits/f7f239cdc17b838ccb856b045355d9c3498bfb9a